<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <include 
        file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name"  value="$(find vitarana_drone)/worlds/task.world"/>
    </include>

    <arg name="record" default="false"/>
	<arg name="duration" default="60"/>
	<arg name="rec_name" default="Task_1_VD.bag"/>

	<group if="$(arg record)">

	    <node name="rosbag_record_position" pkg="rosbag" type="record"
	       args="record -O $(find vitarana_drone)/bag_files/$(arg rec_name) --duration=$(arg duration) /edrone/gps /edrone/gps_velocity" output="screen"/>

	</group>

	<node pkg="vitarana_drone" type="attitude_controller.py" name="attitude_controller" output="screen"/>

	<node pkg="vitarana_drone" type="refactored_drone_controller.py" name="refactored_drone_controller" output="screen"/>
	
	<node pkg="vitarana_drone" type="waypoint_controller.py" name="waypoint_controller" output="screen"/>

	<node pkg="vitarana_drone" type="custom_camera_detect_node.py" name="custom_camera_detect_node" output="screen"/>

    <node pkg="vitarana_drone" type="single_drone_positioning.py" name="single_drone_positioning" output="screen"/>

</launch>
